web/src/components/chat.tsx

```diff
@@ -8,155 +8,147 @@ import type {
+  window.scrollTo({top: document.body.scrollHeight, behavior: 'smooth'})
 }
 
 type ChatParams = {
-  sessionId: string;
-  settings: LLMSettings;
-  onQuery?: (q: string) => any;
-  onNewEntry?: (history: Entry[]) => any;
-};
+  sessionId: string
+  settings: LLMSettings
+  onQuery?: (q: string) => any
+  onNewEntry?: (history: Entry[]) => any
+}
 
-const Chat = ({ sessionId, settings, onQuery, onNewEntry }: ChatParams) => {
-  const [entries, setEntries] = useState<Entry[]>([]);
-  const [current, setCurrent] = useState<CurrentSearch>();
-  const [citations, setCitations] = useState<Citation[]>([]);
+const Chat = ({sessionId, settings, onQuery, onNewEntry}: ChatParams) => {
+  const [entries, setEntries] = useState<Entry[]>([])
+  const [current, setCurrent] = useState<CurrentSearch>()
+  const [citations, setCitations] = useState<Citation[]>([])
 
   const updateCurrent = (current: CurrentSearch) => {
-    setCurrent(current);
-    if (current?.phase === "streaming") {
-      scroll30();
+    setCurrent(current)
+    if (current?.phase === 'streaming') {
+      scroll30()
     }
-  };
+  }
 
-  const updateCitations = (
-    allCitations: Citation[],
-    current?: CurrentSearch
-  ) => {
-    if (!current) return;
+  const updateCitations = (allCitations: Citation[], current?: CurrentSearch) => {
+    if (!current) return
 
-    const entryCitations = Array.from(current.citationsMap.values());
+    const entryCitations = Array.from(current.citationsMap.values())
     if (!entryCitations.some((c) => !c.index)) {
       // All of the entries citations have indexes, so there weren't any changes since the last check
-      return;
+      return
     }
 
     // Get a mapping of all known citations, so as to reuse them if they appear again
-    const citationsMapping = Object.fromEntries(
-      allCitations.map((c) => [c.title + c.url, c.index])
-    );
+    const citationsMapping = Object.fromEntries(allCitations.map((c) => [c.title + c.url, c.index]))
 
     entryCitations.forEach((c) => {
-      const hash = c.title + c.url;
-      const index = citationsMapping[hash];
+      const hash = c.title + c.url
+      const index = citationsMapping[hash]
       if (!index) {
-        c.index = allCitations.length + 1;
-        allCitations.push(c);
+        c.index = allCitations.length + 1
+        allCitations.push(c)
       } else {
-        c.index = index;
+        c.index = index
       }
-    });
-    setCitations(allCitations);
-    setCurrent(current);
-  };
+    })
+    setCitations(allCitations)
+    setCurrent(current)
+  }
 
   const addEntry = (entry: Entry) => {
     setEntries((prev) => {
-      const entries = [...prev, entry];
+      const entries = [...prev, entry]
       if (onNewEntry) {
-        onNewEntry(entries);
+        onNewEntry(entries)
       }
-      return entries;
-    });
-  };
+      return entries
+    })
+  }
 
   const search = async (
```

Comment:
this function has a couple of changes - it accepts different arguments and has an AbortController + some changes in the handling of the search result